src/ert/gui/simulation/view/progress_widget.py

```diff
@@ -89,17 +90,30 @@ def repaint_components(self) -> None:
     def stop_waiting_progress_bar(self) -> None:
         self._waiting_progress_bar.setVisible(False)
 
-    def start_waiting_progress_bar(self) -> None:
-        self._waiting_progress_bar.setVisible(True)
-
     def update_progress(self, status: dict[str, int], realization_count: int) -> None:
         self._status = status
         self._realization_count = realization_count
-        if status.get("Finished", 0) < self._realization_count:
-            self.start_waiting_progress_bar()
-        else:
-            self.stop_waiting_progress_bar()
         self.repaint_components()
 
+    def _update_waiting_progress_bar(self) -> None:
+        self._waiting_progress_bar.setVisible(
+            self._status.get("Finished", 0) < self._realization_count
+        )
+
+    def set_ensemble_failed(self) -> None:
+        for label in self._progress_label_map.values():
+            label.setFixedWidth(0)
+            label.setVisible(False)
+
+        failed_label = self._progress_label_map[REALIZATION_STATE_FAILED]
+        failed_label.setFixedWidth(self.width())
+        failed_label.setVisible(True)
+        self.stop_waiting_progress_bar()
+
+    def set_ensemble_running(self) -> None:
+        for label in self._progress_label_map.values():
+            label.setFixedWidth(0)
+            label.setVisible(False)
```

Comment:
This could have been extracted as a separate function?
But I also wonder why width is set to zero, if the label is hidden? I'm not sure if this has any effect.

Reply:
This is how it was done in the constructor when initially creating the labels. When rerunning, I want it as similar to how it was before the first run as possible. 
I will extract it to a separate function and use it in the constructor as well 👍

Reply:
^it is apparently not how it is done in the constructor...
I think that when we rerun, the first label to be updated can be `REALIZATION_SUCCESS`, and if we already have a invisible `REALIZATION_FAILED` label already filling some space, it can mess up the layout. 

Do invisible widgets take up space? If not, we can remove the `label.setFixedWidth(0)`.

Reply:
My thought was no, but ChatGPT says otherwise. I would test this myself if I were you.

Reply:
It should be fine, the scenario is already handled in `repaint_components`.

Reply:
I will remove the `setFixedWidth(0)`